### DISCUSSION TITLE
ImageMagick fails to load jpeg.la

### DISCUSSION THREAD

**Discussion by acc987** [ImageMagick/ImageMagick#7781] [Help] [answered]
### ImageMagick version

7.1.1-40

### Operating system, version and so on

openSUSE Leap 15.6

### Description

Hey,

I did update ImageMagick after you managed to resolve an issue I had with an older version, but am now encountering a new problem. When trying to read a JPEG image with `identify` I am now getting the following error:
```
magick: unable to load module '/usr/lib64/ImageMagick-7.1.1/modules-7_Q16HDRI10/coders/jpeg.la': file not found @ error/module.c/OpenModule/1293.
magick: no decode delegate for this image format `JPEG' @ error/constitute.c/ReadImage/746.
```
Weirdly enough, the file does exist at the specified location and is readable (and even executable, though it appears to be a text file) for all users. Is anyone else having this issue or knows how I could investigate this?

Thanks in advance for your help!

**Comment by fmw42** [comment]
Post the file

**Reply by acc987** [reply]
This happens with every JPEG file, so I won't be posting an example.

**Reply by fmw42** [reply]
What do you get from 

`magick -version`

and 

`magick -list format`

for the line for JPG/JPEG

**Reply by acc987** [reply]
The former gives:
```
Version: ImageMagick 7.1.1-40 Q16-HDRI x86_64 22476 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules 
Delegates (built-in): bzlib cairo djvu fftw fontconfig freetype heic jbig jng jp2 jpeg lcms lqr ltdl lzma openexr pangocairo png raw rsvg tiff webp wmf x xml zlib zstd
Compiler: gcc (7.5)
```
The output of the latter does not include a line with JPG or JPEG.

I did find out what's wrong though, see my answer below.

**Comment by acc987** [comment, marked as answer]
I am not a linux developer, but I did dive into the MagickCore/module.c code and it seems that it's a `lt_dlopen()` call that fails. Apparently that function belongs to GNU's libtldl/libtool and is used to load dynamic libraries such as ImageMagick's jpeg.so. When turning on appropriate debug output via `export LD_DEBUG=all` as described in https://stackoverflow.com/a/9567335, I eventually came across:
```
/usr/lib64/ImageMagick-7.1.1/modules-7_Q16HDRI10/coders/jpeg.so: error: symbol lookup error: undefined symbol: jpeg16_write_scanlines, version LIBJPEG_8.0 (fatal)
```

As it turned out, the dependencies of my ImageMagick package were not specified correctly, it was happy with libjpeg8 8.2 even though it actually required version 8.3. (Unfortunately this is not the first time I ran into dependency issues due to incorrect package specifications with openSUSE Leap packages.)

After updating to libjpeg8 8.3.3 everything works again. (Sorry for bothering you in the first place, but I wasn't sure where exactly the issue was coming from.)
